src/Microsoft.DotNet.Darc/src/Darc/Operations/GetHealthOperation.cs

```diff
@@ -0,0 +1,291 @@
+                        {
+                            outputBuilder.AppendLine($"    {dependency.Name}");
+                        }
+                    }
+
+                    if (healthMetric.DependenciesThatDoNotFlow.Any())
+                    {
+                        outputBuilder.AppendLine($"  Dependencies that do not flow automatically (disabled or frequency=none):");
+                        foreach (DependencyDetail dependency in healthMetric.DependenciesThatDoNotFlow)
+                        {
+                            outputBuilder.AppendLine($"    {dependency.Name}");
+                        }
+                    }
+
+                    if (healthMetric.UnusedSubscriptions.Any())
+                    {
+                        outputBuilder.AppendLine($"  Subscriptions that do not have any effect:");
+                        foreach (Subscription subscription in healthMetric.UnusedSubscriptions)
+                        {
+                            outputBuilder.AppendLine($"    {UxHelpers.GetSubscriptionDescription(subscription)} ({subscription.Id})");
+                        }
+                    }
+
+                    return new HealthMetricWithOutput(healthMetric, outputBuilder.ToString());
+                })
+            .ToList();
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="channels"></param>
+        /// <returns></returns>
+        private HashSet<string> ComputeChannelsToEvaluate(IEnumerable<Channel> channels)
+        {
+            if (!string.IsNullOrEmpty(_options.Channel))
+            {
+                HashSet<string> channelsToTarget = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+                Channel targetChannel = UxHelpers.ResolveSingleChannel(channels, _options.Channel);
+
+                if (targetChannel != null)
+                {
+                    channelsToTarget.Add(targetChannel.Name);
+                }
+
+                return channelsToTarget;
+            }
+            else
+            {
+                // Look up all channels
+                return channels.Select(c => c.Name).ToHashSet();
+            }
+        }
+
+        /// <summary>
+        ///     Compute maximal initial set of input repositories
+        /// </summary>
+        /// <param name="remote">BAR remote</param>
+        /// <returns>Repositories to evaluate</returns>
+        private HashSet<string> ComputeRepositoriesToEvaluate(IEnumerable<DefaultChannel> defaultChannels, IEnumerable<Subscription> subscriptions)
+        {
+            // Compute which repositories to target
+            HashSet<string> reposToTarget = defaultChannels
+                .Where(df => string.IsNullOrEmpty(_options.Repo) || df.Repository.Contains(_options.Repo, StringComparison.OrdinalIgnoreCase))
+                .Select(df => df.Repository)
+                .ToHashSet(StringComparer.OrdinalIgnoreCase);
+
+            subscriptions
```

Comment:
[nit] Curious on why you picked this approach over `reposToTarget.UnionWith(subscriptions.Where(...)`